hb-depend does not build with HB_NO_MATH or HB_NO_COLOR

```
In file included from harfbuzz/src/harfbuzz-subset.cc:10:
harfbuzz/src/hb-depend.cc:86:15: error: no member named 'MATH' in 'hb_ot_face_t'
   86 |   face->table.MATH->depend (this);
      |   ~~~~~~~~~~~ ^
harfbuzz/src/hb-depend.cc:87:15: error: no member named 'COLR' in 'hb_ot_face_t'
   87 |   face->table.COLR->depend (this);
      |   ~~~~~~~~~~~ ^
In file included from harfbuzz/src/harfbuzz-subset.cc:15:
In file included from harfbuzz/src/hb-font.cc:38:
```

--- a/.github/workflows/configs-build.yml
+++ b/.github/workflows/configs-build.yml
@@ -33,7 +33,12 @@ jobs:
         #include "src/harfbuzz.cc"
         int main() { return 0; }
         EOF
+        cat <<EOF > subset.cc
+        #include "src/harfbuzz-subset.cc"
+        EOF
     - name: install dependencies
       run: sudo apt-get install gcc
     - name: Build with ${{ matrix.config }}
       run: g++ -std=c++11 main.cc -D${{ matrix.config }}
+    - name: Build subset library with ${{ matrix.config }}
+      run: g++ -std=c++11 -c subset.cc -D${{ matrix.config }}

--- a/src/hb-depend.cc
+++ b/src/hb-depend.cc
@@ -83,10 +83,16 @@ hb_depend_data_builder_t::compile (hb_face_t *face)
    * constructor (which references a hidden is_blocklisted symbol) in this TU. */
   if (hb_ot_layout_has_substitution (face))
     face->table.GSUB.get_relaxed ()->depend (this, face);
+#ifndef HB_NO_MATH
   face->table.MATH->depend (this);
+#endif
+#ifndef HB_NO_COLOR
   face->table.COLR->depend (this);
+#endif
   face->table.glyf->depend (this);
+#ifndef HB_NO_CFF
   OT::cff1_subset_accelerator_t (face).depend (this);
+#endif
   /* XXX TODO: add face->table.VARC->depend (this) here.
    * VARC closure and subsetting are not yet implemented (see hb-subset-plan.cc),
    * so the right traversal architecture for VarComponent records hasn't been

